### DISCUSSION TITLE
How to fix - Payload Too Large

### DISCUSSION THREAD

**Discussion by kevinh94** [strapi/strapi#7767] [Legacy Topics]
Hi,

I have Strapi running behind an NGINX server and I POST requests that are around 200Ko. (I POST the HTML source code of a page in the data field of an "HTML Pages" Content Type). I keep getting the 413 Error - Payload Too Large from Strapi.

I have added the following to nginx.conf (both on the 80 and 443 ports):
```sh
max_client_body_size 20M;
```
However, Strapi debug logs indicate that requests go through NGINX to Strapi successfully. So I don't think NGINX is the problem here.

I have also tried to create the folders and file config/environments/development/request.json with this configuration in it:
```json
{
    "parser": {
        "enabled": true,
        "multipart": true,
        "formLimit": "500mb",
        "jsonLimit": "500mb",
        "formidable": {
            "maxFileSize": 524288000
        }
    }
}
```
I am in a development environment but, just in case I was not, I added the config/environments/production/request.json file as well.

However, I keep getting the same 413 error. After going through the current documentation, it seems to me that this request.json worked on a former version of Strapi but not this one. 

So here's my question. How do you guys increase the maximum payload size in Strapi? (I have the 3.1.4 version).

Thanks in advance for your help! :)

**Comment by derrickmehaffy** [comment]
If you are on the Strapi stable releases (`v3.0.x` or `v.3.1.x`) you will need to switch to the new configuration structure, specifically for parser it will be either `./config/middleware.js` or `./config/env/*/middleware.js` using the following docs: 

https://strapi.io/documentation/v3.x/concepts/middlewares.html
https://strapi.io/documentation/v3.x/concepts/middlewares.html#request-middlewares

The `.json` files and likewise `./config/environments/production/request.json` are from the beta and won't work in the newer versions. I assume you found these with a Google search, and our new docs haven't been entirely indexed by Google yet.

**Reply by kevinh94** [reply]
Thank you so much, I edited ./config/middleware.json and added the parser field mentioned above and it worked!

Have a great day :)

**Comment by sanjeev081996** [comment]
I have strapi running locally and try to upload video using baseencode 64, but I got the error of 413 whenever my file size is larger than the 1mb. I have created the request.json file inside the ./config/request.json and as well as middleware mentioned in the documentation, but it didn't help me
request.json code mentioned below
```json
{
	"parser": {
		"enabled": true,
		"multipart": true,
		"formLimit": "5242880000",
		"jsonLimit": 5242880000,
		"formidable": {
			"maxFileSize": 524288000
		}
	}
}
```
middleware.json

```js
module.exports = {
  load: {
   after: ['parser'],
  },
};
```

config directory 
![image](https://user-images.githubusercontent.com/56131320/94889263-2a3bc680-0499-11eb-853c-37e92c9a9592.png)

**Reply by derrickmehaffy** [reply]
> error of 413 whenever my file size is larger than the 1mb

I believe that issue would be related to the use of a proxy, as it sounds like the default limit of Nginx, can you try to add in the `client_max_body_size` to your Nginx Virtual host config if you are using Nginx? 

http://nginx.org/en/docs/http/ngx_http_core_module.html#client_max_body_size
